### DISCUSSION TITLE
Autofill part quantity when purchasing

### DISCUSSION THREAD

**Discussion by SebBorbely** [inventree/InvenTree#4903] [Q&A] [answered]
I am having a small issue with how inventree auto fills parts for purchase orders.
Example: I have a part that is required for build orders and sales orders. I have 45 in stock and have allocated those to the build orders and need an additional 15 to complete the build orders. As well as this I also have a purchase order that requires 3 more of these parts.
I have already added 15 of these parts to a PO however to complete the orders I need to buy in 3 more of this part.

![Capture](https://github.com/inventree/InvenTree/assets/99136411/d93df956-fbea-42d2-adaa-8fbd4388308f)

I click on the shopping cart doesn't matter if that's in the part page, the BO page or the SO page it all acts the same way. This opens up the following box with the part.

![Capture1](https://github.com/inventree/InvenTree/assets/99136411/bc2ac347-c903-478c-bc60-c35db185e9ce)

And here we come to my issue. The quantity to order gets auto filled to 18. That is the value needed to complete the BO and SO so its wonderful that it auto calculates that for me however it does not take into account that I already have 15 of those parts on order so I only need 3 more of them. If I didn't take note of this it would cause me to over order this part.

(An additional note that is not present in this example but this part is also an assembly so I can also issue a build order for it. If I set up a BO for 10 it still doesn't change how many it auto fills for a PO. )

So after all that my question is this. Is there a way for this auto allocation to take into account what's already on order or what's already getting built? So in the above example it would autofill to 3 instead of 18. Is this already something doable and I am missing it or is this something that is in future plans? I think this would be quite a useful addition unless I'm missing a reason for it not to work like this.

**Comment by SchrodingersGat** [comment, marked as answer]
@SebBorbely thanks for raising this - I have created a new issue here - https://github.com/inventree/InvenTree/issues/4904

This does need some attention.
